modules/perception/lidar/lib/segmentation/ncut/ncut_segmentation.cc

```diff
@@ -218,25 +219,29 @@ bool NCutSegmentation::Segment(const SegmentationOptions& options,
     AINFO << "remove ground";
     cloud_above_ground->CopyPointCloud(*lidar_frame_ref_->cloud,
                                        lidar_frame_ref_->non_ground_indices);
+  } else {
+    // if used as secondary segmentor, got from cloud directly
+    cloud_above_ground->CopyPointCloud(*lidar_frame_ref_->cloud,
+                                       lidar_frame_ref_->secondary_indices);
   }
 
+#ifdef DEBUG_NCUT
   // filter_by_ground(cloud, non_ground_indices, &cloud_above_ground);
-  AINFO << "filter ground, elapsed time: " << omp_get_wtime() - start_t;
+  AINFO << "filter ground, elapsed time: " << omp_get_wtime() - start_t
+        << cloud_above_ground->size() << " points left";
   start_t = omp_get_wtime();
-#ifdef DEBUG_NCUT
-  AINFO << "filter ground: " << cloud_above_ground->size() << " points left";
   VisualizePointCloud(cloud_above_ground);
 #endif
 
   // .3 filter vehicle
   base::PointFCloudPtr cloud_after_car_filter;
   ObstacleFilter(cloud_above_ground, vehicle_filter_cell_size_, false,
                  &cloud_after_car_filter, segments);
-  AINFO << "filter vehicle, elapsed time: " << omp_get_wtime() - start_t;
-  start_t = omp_get_wtime();
 #ifdef DEBUG_NCUT
-  AINFO << "filter vehicle: " << cloud_after_car_filter->size()
+  AINFO << "filter vehicle, elapsed time: " << omp_get_wtime() - start_t
```

Comment:
Same. Use ADEBUG when necessary.

Reply:
same here. it is protected by compilation flag.